build-logic/build.gradle.kts

```diff
@@ -32,8 +32,17 @@ repositories {
 dependencies {
     // Needed for caching reflected data during builds
     implementation("org.terasology:reflections:0.9.12-MB")
-    implementation("org.javassist:javassist:3.27.0-GA")
-    implementation("dom4j:dom4j:1.6.1")
+
+    // Corrections for old reflections dependencies:
+    modules {
+        module("dom4j:dom4j") {
+            replacedBy("org.dom4j:dom4j")
```

Comment:
okay, fair enough, but before we had pinned the version, so why don't we pin it anymore now?

Reply:
ah, right, because the reason I did that `replacedBy` in the first place is because the newer versions are under the `org.dom4j` group.

Then when I went to see how that changed the [dependency graph][scan], I found no dom4j at all:

- org.terasology:reflections:0.9.12-MB
  + com.google.guava:guava:20.0 → 31.1-jre
  + org.javassist:javassist:3.21.0-GA → 3.29.0-GA

[scan]: https://scans.gradle.com/s/prrofsqgzegjs/dependencies?focusedDependency=WzAsMCwxMCxbMCwwLFsxMF1dXQ&focusedDependencyView=dependencies_or_failure&toggled=W1swXSxbMCwwXSxbMCwwLFsxMF1dLFswLDAsWzEwLDExXV1d

As far as I can tell, reflections considered dom4j to be an optional dependency, so when it couldn't find its originally required version anymore, it gave up? I'm not 100% sure how that works, I'm not so fluent in maven and optional dependencies.

and now that we don't do compile-time scans, reflections is fine without the DOM library because we don't have it writing XML anymore?

I know that might be less than confidence-boosting but I'm trying not to think too much about [the state of the Reflections library](https://github.com/MovingBlocks/gestalt/issues/132#issuecomment-1132166118).

Reply:
okay, then follow-up question: if it's fine without the DOM library, then why do we still need it at all? :sweat_smile:

Reply:
hmm, yes. _<checks again>_

wait, it wasn't the `replaceBy` that made it go away, it was the not-requiring-it that made it go away! that makes sense.